File: app/src/main/java/com/example/hellooboe/NativeMethods.java
package com.example.hellooboe;

public class NativeMethods {

    public static native void createStream();

    public static native void destroyStream();

    public static native void startStream();

    public static native void stopStream();

    public static native boolean audioDataIsFloat();

    public static native double getMinAmp();

    public static native double getMaxAmp();

    public static native double getMinFreq();

    public static native double getMaxFreq();

    public static native float[] loadConstant(double start, double length, int row, int col, int width);

    public static native void makeSound();

    public static native double getAmpTime(int index, int samplesPerSecond);

    public static native double getFreqTime(int index, int samplesPerSecond);

    public static native int getSampleRate();

    public static native void setData(int row, int col, double start, double end, double length, double cycles, double min, double max);

    /*
    public static native float[] loadExponential(double start, double end, double length, int row, int col);

    public static native float[] loadLinear(double start, double end, double length, int row, int col);

    public static native float[] loadLogarithm(double start, double end, double length, int row, int col);

    public static native float[] loadNthRoot(double start, double end, double length, int row, int col);

    public static native float[] loadPower(double start, double end, double length, int row, int col);

    public static native float[] loadQuadratic(double start, double end, double length, int row, int col);

    public static native float[] loadSine(double start, double end, double length, int row, int col);
     */


}

File: app/src/main/java/io/fourth_finger/sound_sculptor/NativeMethods.kt
package io.fourth_finger.sound_sculptor

object NativeMethods {

    external fun createStream()


    external fun audioDataIsFloat(): Boolean

    external fun destroyStream()
    external fun startStream()
    external fun stopStream()
    val minAmp: Double
        external get
    val maxAmp: Double
        external get
    val minFreq: Double
        external get
    val maxFreq: Double
        external get

    external fun generateConstant(
        start: Double,
        length: Double,
        row: Int,
        col: Int,
        width: Int
    ): FloatArray?

    external fun makeSound()
    external fun getAmpTime(index: Int, samplesPerSecond: Int): Double
    external fun getFreqTime(index: Int, samplesPerSecond: Int): Double
    val sampleRate: Int
        external get

    external fun setData(
        row: Int,
        col: Int,
        start: Double,
        end: Double,
        length: Double,
        cycles: Double,
        min: Double,
        max: Double
    ) /*
    public static native float[] loadExponential(double start, double end, double length, int row, int col);

    public static native float[] loadLinear(double start, double end, double length, int row, int col);

    public static native float[] loadLogarithm(double start, double end, double length, int row, int col);

    public static native float[] loadNthRoot(double start, double end, double length, int row, int col);

    public static native float[] loadPower(double start, double end, double length, int row, int col);

    public static native float[] loadQuadratic(double start, double end, double length, int row, int col);

    public static native float[] loadSine(double start, double end, double length, int row, int col);
     */
}